kermit/src/commonMain/kotlin/co/touchlab/kermit/Logger.kt

```diff
@@ -10,26 +10,163 @@
 
 package co.touchlab.kermit
 
-abstract class Logger{
-    open fun isLoggable(severity: Severity): Boolean = true
+open class Logger(
+    val config: LoggerConfig,
+    val tag: String = config.defaultTag
+) {
+    fun withTag(tag: String): Logger {
+        return Logger(this.config, tag)
+    }
 
-    abstract fun log(severity: Severity, message: String, tag: String, throwable: Throwable? = null)
+    /**
+     * We could reduce the methods on this object, but the native objc
+     * export can't use default arguments, so we have a few extra methods defined.
+     */
+    inline fun v(message: () -> String) {
+        if (config.minSeverity <= Severity.Verbose)
+            log(Severity.Verbose, tag, null, message())
+    }
 
-    open fun v(message: String, tag: String, throwable: Throwable? = null) =
-        log(Severity.Verbose, message, tag, throwable)
+    inline fun v(throwable: Throwable, message: () -> String) {
+        if (config.minSeverity <= Severity.Verbose)
+            log(Severity.Verbose, tag, throwable, message())
+    }
 
-    open fun d(message: String, tag: String, throwable: Throwable? = null) =
-        log(Severity.Debug, message, tag, throwable)
+    fun v(message: String, throwable: Throwable? = null) {
```

Comment:
the flipped param order of this method and the one above it is a little weird. i assume its to take advantage of default args and trailing lamba respectively?

Reply:
This whole class is going to be confusing because "Logger" is now "LogWriter" and "Kermit" is now "Logger", so they're not really the same thing. The param order is for default params, and you can no longer override the tag without creating a new instance of `Logger`. Tags are very android-centric, and having them as params made the Swift side painful, so I got rid of them as an option. Can discuss if you're still not sure what's happening here.